File: src/main/java/net/benwoodworth/fastcraft/core/FastCraft.java
package net.benwoodworth.fastcraft.core;

import net.benwoodworth.fastcraft.core.api.dependencies.Dependencies;
import org.jetbrains.annotations.NotNull;

/**
 * The core class of FastCraft.
 */
public class FastCraft {

    @NotNull
    private Dependencies dependencies;

    /**
     * Create a new instance of FastCraft.
     *
     * @param dependencies FastCraft dependencies.
     */
    public FastCraft(@NotNull Dependencies dependencies) {
        this.dependencies = dependencies;
    }

    /**
     * Get the FastCraft dependencies.
     *
     * @return Returns FastCraft dependencies.
     */
    @NotNull
    public Dependencies getDependencies() {
        return dependencies;
    }
}

File: src/main/java/net/benwoodworth/fastcraft/core/FastCraft.kt
package net.benwoodworth.fastcraft.core

import net.benwoodworth.fastcraft.core.api.FastCraftApi
import net.benwoodworth.fastcraft.core.api.inventory.InventoryProvider

abstract class FastCraft(
        val logger: ConsoleLogger,
        val inventoryProvider: InventoryProvider<*> //,
//        val api: FastCraftApi,
//
//        val configProvider: ConfigProvider,
//        val lang: LangProvider
) {

    val api = FastCraftApi(this)

    abstract fun pluginStart()

//    fun pluginStop()
//
//    fun pluginReload()

}

